File: app/src/main/java/example/prada/lab/pradaoutlook/model/IEventStore.java
package example.prada.lab.pradaoutlook.model;

import android.support.annotation.NonNull;

import java.util.Calendar;
import java.util.List;

/**
 * Created by prada on 10/30/16.
 */

public interface IEventStore {

    @NonNull List<POEvent> queryEvents(long t1, long t2);

    int countEvents(long t1, long t2);

    boolean hasEvents(long t1, long t2);

    Calendar getFirstEventTime();

    Calendar getLatestEventTime();
}

File: app/src/main/java/example/prada/lab/pradaoutlook/store/IEventStore.kt
package example.prada.lab.pradaoutlook.store

import android.database.Cursor
import example.prada.lab.pradaoutlook.model.IEventDataUpdatedListener
import example.prada.lab.pradaoutlook.model.POEvent
import java.util.Calendar

interface IEventStore {
    fun getEvents(): Cursor
    fun countEvents(): Int
    fun countEvents(t1: Long, t2: Long): Int
    fun getFirstEventTime(): Calendar
    fun getLatestEventTime(): Calendar
    fun addEvents(events: Collection<POEvent>?)
    fun addListener(listener: IEventDataUpdatedListener?)
    fun removeListener(listener: IEventDataUpdatedListener?)
    fun removeAllRecords()
}
